incubator/hnc/api/v1alpha1/hnc_config.go

```diff
@@ -99,6 +109,36 @@ type HNCConfigurationList struct {
 	Items           []HNCConfiguration `json:"items"`
 }
 
+// HNCConfigurationCode is the machine-readable, enum-like type of `HNCConfigurationCondition.Code`.
+// See that field for more information.
+type HNCConfigurationCode string
+
+// HNCConfigurationCondition specifies the code and the description of an error condition.
+type HNCConfigurationCondition struct {
+	// Describes the condition in a machine-readable string value. The currently valid values are
+	// shown below, but new values may be added over time. This field is always present in a
+	// condition.
+	//
+	// All codes that begin with the prefix `crit` indicate that reconciliation has
+	// been paused for this configuration. Future changes of the configuration will be
+	// ignored by HNC until the condition has been resolved. Non-critical conditions
+	// typically indicate some kinds of error that HNC itself can ignore. However,
+	// the behaviors of some types might be out-of-sync with the users' expectations.
+	//
+	// Currently, the supported values are:
+	//
+	// - "critSingletonNameInvalid": the specified singleton name is invalid. The name should be the
```

Comment:
Maybe: "An HNCConfiguration object exists with a name other than `config`. The object will be identified in the message."

Reply:
I didn't change this documentation because we probably cannot write the error message in the correct singleton. See my comment reply below.